### ISSUE TITLE
CPUfreq bypass not working and CPU scaling is disabled.

### ISSUE THREAD

**Issue by jeois** [Frogging-Family/linux-tkg#507]
My intel_pstate _passive_ governor bypass isn't working, so my CPUs fall back to the "powersave" governor setting. In a normal "powersave" governor, it significantly reduces boost, but in this case it seems to completely disable CPU scaling altogether and all cores are **stuck at 800mhz**.    Needless to say, performance is awful.

I'm able to get CPU frequency scaling back by using `sudo cpupower frequency-set -g ondemand`, but I'm afraid it's not using the more aggressive`ACPI_cpufreq` driver's "ondemand" governor.  I enabled cpupower's systemd unit, but it's not applying the setting on boot (not important).  The problem is this setting switches the governors, but I think I'm still only using `intel_pstate`'s governor rather than as passthrough for `ACPI_cpufreq`.

I was sure this was my fault, as I must have screwed up the configuration somehow:
[customization.cfg.txt](https://github.com/Frogging-Family/linux-tkg/files/8597670/customization.cfg.txt)
[prepare.log.txt](https://github.com/Frogging-Family/linux-tkg/files/8597671/prepare.log.txt)
[system-info.txt](https://github.com/Frogging-Family/linux-tkg/files/8597672/system-info.txt)

Since I'm not using standard base modules, I wouldn't have bothered asking since that's clearly my fault and I can slowly add them back, but when I installed `chaotic-aur`'s latest binary build `linux-tkg-bmq-generic_v3 (5.17.5-256)`, it has the same issue of the CPUs stuck at 800mhz until I set the governor to "ondemand" manually.  They obviously are using the base modules config, so now I'm really confused.  Here's their customization.cfg from _/usr/share/doc/linux-tkg-bmq-generic_v3/_
[customization.cfg.txt](https://github.com/Frogging-Family/linux-tkg/files/8597713/customization.cfg.txt)

Is there a way to check if `intel_pstate` is in passive mode?  because I tried another kernel from a different source, and it seems to be working fine even though `cpupower` is indicating "powersave".  (they probably are using a similar CPUfreq bypass) 
How do I query the status of `ACPI_cpufreq`?  with this bypass setup, I don't know how to tell if it's running or working.  What are the prerequisites for this CPUfreq bypass to work in the first place?  I tried building `ACPI_cpufreq` as a module and into the kernel directly, and it doesn't work either way.

**Comment by jeois** [comment]
btw, I'm using intel i7-9700.  (hardware info wasn't in the logs)

This issue is more persistent than I initially thought.  No matter what I do, `intel_pstate` refuses default to anything other than the "powersave" governor.  And when running a tkg kernel, that also locks the clocks to only 800mhz.  I tried using `tlp` and `cpupower` and `cpufrequtils` to change it during boot, but it freezes or reverts to "powersave" when rebooted. 

I don't get why manually changing governors with the `cpupower` command works, but its systemd unit refuses to set it at boot.  I don't have any other power management or CPU frequency software that I'm aware of; this is a deskop, not a laptop after all.

If anyone has ideas how to set the CPU governor permanently, or better yet, the perquisites for getting `ACPI_cpufreq` to work instead, please let me know.  Thanks in advance.

**Comment by jeois** [comment]
I got the correct driver running with `modprobe acpi-cpufreq`.   It was as simple as loading it ontop of 'intel_pstates'.  For reference, you can tell `intel_pstates` is in passive mode by querying `cpupower frequency-info` and if it says "driver: intel_**cpufreq**" instead of "intel_pstates" then it should be bypassing it for ACPI-CPUfreq.  So, that's one question answered.

The second issue was solved by `systemctl mask power-profile-daemon`.  There was a recent update to `power-profile-daemon`, so if you're using it with KDE or Gnome settings, it'll conflict with cpupower.  After which, I enabled the `cpupower` systemd script again, and now it'll set the correct CPUfreq governor during boot based on `/etc/default/cpupower` config.

So, I managed to get my system working... However, the issue persists since I can tell part of my boot time is un-scaled (no cpu frequency steps/boosts), making the loading process fairly slow.  I still have no idea why it's defaulting to intel_pstate's "powersave" or why tkg kernels using that particular governor locks everything to 800mhz for extra slowness.

**Comment by Tk-Glitch** [comment]
> In a normal "powersave" governor, it significantly reduces boost, but in this case it seems to completely disable CPU scaling altogether and all cores are stuck at 800mhz.

Depends on what you call "normal". Only intel_pstate powersave governor allows for frequency scaling. Acpi_cpufreq powersave locks frequency to the lowest frequency supported by the CPU. So passive mode is working as it should here.

The problem seems to be that your OS enforces powersave on boot, bypassing the tkg default ondemand. It's probably something you have installed to manage power since it's a laptop or a flag set in your bootloader as you seem to also experience slow boot, and it's not an actual kernel issue.

I also can't reproduce any of this on any of my Intel CPUs, be it mobile or desktop. The ondemand governor is selected by default on my end. That being said, I'm not using any power management daemon, unlike you.

**Comment by jeois** [comment]
Thanks for the response and the info on the governors.

I rather not use power management daemons, and I only started using `cpupower` because of this issue.  I didn't know `power-profile-daemon` was installed.  I suspect a recent update was the culprit, but the issue persists after I disabled it.  I just checked again with both of them disabled, and it was back to 800mhz powersave.  I only briefly used the other daemons to try to automate setting it back to 'ondemand' during boot or see if they expose a setting I didn't know about.

You're probably right though.  It might not be the kernel or its configuration.  Thanks anyway.
